### PR TITLE
vimc-4457: Add bundle pack/import for remotes

### PR DESCRIPTION
**Pull request by richfitz**
This PR completes the remote support for bundles, by adding two new functions to orderly for packing and importing from a remote.

### CODE REVIEW

File: R/bundle.R [hunk 1]
```diff
@@ -18,8 +18,8 @@
 ##' \item{\code{orderly_bundle_run}}{The path to the packed bundle (a zip
 ##'   file created by \code{orderly_bundle_pack})}
 ##'
-##' \item{\code{orderly_bundle_run}}{The path to unpack and the run
-##'   the bundle (a zip file created by \code{orderly_bundle_run}}
+##' \item{\code{orderly_bundle_import}}{The path to unpack and import
+##'   (a zip file created by \code{orderly_bundle_run}}
```
Comment (mwoodbri) [inline R/bundle.R:22]:
```suggestion
##'   (a zip file created by \code{orderly_bundle_run})}
```

File: R/remote.R [hunk 2]
```diff
@@ -496,3 +496,72 @@ remote_report_update_metadata <- function(name, remote, config) {
   }
   data_frame(id = ids, path = file.path(path_cache, ids))
 }
+
+
+##' Pack a bundle on a remote. This is like calling
+##' \code{\link{orderly_bundle_pack}} on the remote and can be used to
+##' extract a long-running report from a server to run (say) on a HPC
```
Comment (mwoodbri) [inline R/remote.R:503]:
```suggestion
##' extract a long-running report from a server to run (say) on an HPC
```

File: R/remote.R [hunk 3]
```diff
@@ -496,3 +496,72 @@ remote_report_update_metadata <- function(name, remote, config) {
...
+##' system.
+##'
+##' The workflow here will typically be:
+##'
+##' 1. Use \code{orderly_bundle_pack_remote()} to create a local
+##'    copy of a bundle, extracted from a remote. Typically this will
+##'    be run fro the system where the bundle will be run (an HPC
```
Comment (mwoodbri) [inline R/remote.R:510]:
```suggestion
##'    be run from the system where the bundle will be run (an HPC
```

File: R/remote.R [hunk 4]
```diff
@@ -496,3 +496,72 @@ remote_report_update_metadata <- function(name, remote, config) {
...
+##'    head-node or a other powerful computer).
```
Comment (mwoodbri) [inline R/remote.R:511]:
```suggestion
##'    head-node or another powerful computer).
```

File: tests/testthat/test-remote-path.R [hunk 5]
```diff
@@ -300,3 +300,21 @@ test_that("pull dependencies that use a query", {
     orderly_list_archive(root),
     data_frame(name = "other", id = dat$ids[[2]]))
 })
+
+
+test_that("bundle pack and import", {
+  skip_on_cran_windows()
+  path1 <- prepare_orderly_example("minimal")
+  path2 <- prepare_orderly_example("minimal")
```
Comment (mwoodbri) [inline tests/testthat/test-remote-path.R:308]: Unused

**Review by mwoodbri — CHANGES_REQUESTED** [review]
I haven't run this code but it looks essentially good to my (relatively) untrained eye. Just a couple of typos and a seemingly unused variable that could tidied up.

File: tests/testthat/test-remote-path.R (hunk 5 above)
Comment (richfitz) [reply tests/testthat/test-remote-path.R]: thanks, removed now

**Review by r-ash — APPROVED** [review]
Nice, this looks really useful

**Review by mwoodbri — APPROVED** [review]

**Pull request merged**